apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -460,6 +470,15 @@ type Storage struct {
 	// Whether this disk represents rotational storage
 	Rotational bool `json:"rotational,omitempty"`
 
+	// Device type, one of: HDD, SSD, NVME.
+	// Field value is computed from "Name" and "Rotational" field values.
```

Comment:
Nobody cares how it's computed :)

Reply:
This is for auto-generated docs and CRD comments, should I remove this?

Reply:
Someone looking at the CRD wants to know that it says "NVME" when it's an NVME, "SSD" when it's an SSD, and "HDD" when it's an HDD.
Explaining how it's calculated only introduces a suspicion that it might be wrong. We don't state how any of the other values are calculated.

Reply:
@zaneb Thanks for explaining, recent rebase removed that comment